### ISSUE TITLE
Layout Animation with flatlist numColumns > 1

### ISSUE THREAD

**Issue by Majiedo** [software-mansion/react-native-reanimated#2806]
## Description

Hey i was trying to make layout animation with flatlist it works just fine with numColumns 1 but if i try 2 or more than 1 it doesn't work anymore when i delete item or add new one the whole item re render which not happening when numColumns set to 1

### Expected behavior
when add new item or delete it should animate the other item to re order them not re render the whole list again!.

## Snack or minimal code example

Flatlist -: 
```
<Animated.FlatList
          key={columns ? "oneRow" : "twoRow"}
          data={Tasks.data}
          numColumns={columns ? 1 : 2}
          showsVerticalScrollIndicator={false}
          itemLayoutAnimation={Layout}
          style={styles.Wrapper}
          renderItem={({ item, index }) => (
            <List
              key={index}
              index={index}
              item={item}
              DeleteHanlder={DeleteHanlder}
            />
          )}
        />
```

item itself -: 
```
<Animated.View
        entering={FadeInDown.springify().delay(index * 150)}
        exiting={FadeOutUp.springify()}
        .....
```

## Package versions

- React Native: 0.64.3
- React Native Reanimated: ~2.3.1

## Affected platforms

- [✅] iOS

**Comment by agoyal0422** [comment]
Did you solve this?

**Comment by Majiedo** [comment]
> Did you solve this?

no, i even made repo for this issue if anyone want to clone and test it out [here](https://github.com/Majiedo/TwoColumnsReactNative)

**Comment by agoyal0422** [comment]
Bruh why is this documentation so bad

**Comment by Majiedo** [comment]
> Bruh why is this documentation so bad

sure the documentation not that great but still it's not that bad , but i have been with this issue for like 3 months and still didn't solve it....tried everything but still...i think i missing something so i'm waiting for any better developer to help me 😞

**Comment by Majiedo** [comment]
i found work around if anyone interesting i'll show you guys when i'm back home...

**Comment by Majiedo** [comment]
anyway the solution was to replace flatlist with scrolllist and make the width of the content half of the screen so it will be two columns and add the layout prop to animated view for the item.

**Comment by Latropos** [comment]
Thank you for submitting this issue! Unfortunately we are not planning to support this feature (multiple columns flatList) and I'm glad you've found other solution.

The bug explanation: When `numColumns` is grater than 1 keyExtractor is substituted under the hood, so your keys don't match. It is difficult to fix, since all elements in a single row have the very same key, so you can't differ between two views in a single row.

**Comment by relaxmeister** [comment]
"I'm glad you've found other solution." - replacing flatlist with scrollview is never a good solution for optimizational reasons... it will eventually bite you for that reason.

**Comment by Majiedo** [comment]
> "I'm glad you've found other solution." - replacing flatlist with scrollview is never a good solution for optimizational reasons... it will eventually bite you for that reason.

not really, it depends on the quantity of items within the list.

**Comment by relaxmeister** [comment]
> > "I'm glad you've found other solution." - replacing flatlist with scrollview is never a good solution for optimizational reasons... it will eventually bite you for that reason.
> 
> not really, it depends on the quantity of items within the list.

Absolutely, but im guessing people use flatlist for starters because of the reason it can contain lots of items. If not... im stunned.

**Comment by NiccoloCase** [comment]
I absolutely can't switch to a scrollview for performance reasons! and at the moment the entry and exit animations are broken. Any new solution?

**Comment by Majiedo** [comment]
> I absolutely can't switch to a scrollview for performance reasons! and at the moment the entry and exit animations are broken. Any new solution?

Unfortunately, this is the only workaround I have found to switch the **ScrollView**. I fully understand that the **FlatList** is more optimized for handling larger lists

**Comment by george-raphael** [comment]
I used a library called [moti.fyi](https://moti.fyi) It works like magic :) 

If the list is big, then implement a way to render animation on the first few list items that are on the screen and leave the rest because they are not on the screen anyway :)
